Sources/ClientGtk/ServerListView.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftCrossUI
+import DeltaCore
+
+enum DetailState {
+  case addition, server(ServerDescriptor), error(String)
+}
+
+class ServerListViewState: Observable {
+  @Observed var servers: [ServerDescriptor] = []
+  @Observed var detailState = DetailState.addition
+}
+
+struct ServerListView: View {
+  var completionHandler: (ServerDescriptor) -> Void
+
+  var state = ServerListViewState()
+
+  var body: some ViewContent {
+    NavigationSplitView {
+      VStack {
+        Button("Add server") {
+          state.detailState = .addition
+        }
+
+        ScrollView {
+          ForEach(state.servers) { server in
+            Button(server.name) {
+              state.detailState = .server(server)
+            }
+            .padding(Edge.Set.bottom, 5)
+          }
+        }
+      }
+      .frame(minWidth: 100)
+      .padding(Edge.Set.trailing, 10)
+    } detail: {
+      VStack {
+        switch state.detailState {
+          case .addition:
+            additionView
+          case .server(let server):
+            Text(server.name)
+            if let port = server.port {
+              Text(server.host + ":" + String(port))
+            } else {
+              Text(server.host)
+            }
+
+            Button("Connect") {
+              completionHandler(server)
+            }
+          case .error(let message):
+            Text(message)
```

Comment:
Make this `Text("Error: \(message)")` so that it's more clear that it's an error message.

It would also be good to have a 'back' button that sends you back to the addition view (since that's the only place creating the errors afaict).

Reply:
Sure - I had just left that Text the same as it was before. I did have a Back button but it seemed redundant when the add server button was prominently placed in the top corner but if you think it’s a good idea I’ll readd it.

Reply:
> Sure - I had just left that Text the same as it was before.

Ah right, because the file was renamed it just looked like you wrote it all (couldn't see what was copied from the other file and what was written by you).

Yeah the main reason is that when you open it with no servers it automatically brings you to the addition view, so it's slightly weird having to click the add button to go back when you didn't do it in the first place.